packages/suite/src/components/suite/hocNotification/components/withConditionalAction.tsx

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import styled from 'styled-components';
+import { connect } from 'react-redux';
+import { useRouteMatch } from 'react-router-dom';
+
+import { useSelector } from '@suite-hooks';
+import { variables } from '@trezor/components';
+
+import type { NotificationEntry } from '@suite-reducers/notificationReducer';
+import type { Dispatch, Action, ExtendedMessageDescriptor } from '@suite-types';
+import type { Network } from '@wallet-types';
+import type { ViewProps } from '../definitions';
+
+const Header = styled.div`
+    font-weight: ${variables.FONT_WEIGHT.MEDIUM};
+    color: ${props => props.theme.TYPE_LIGHT_GREY};
+    margin-top: 1px;
+`;
+
+const Body = styled.div`
+    font-weight: ${variables.FONT_WEIGHT.MEDIUM};
+    margin-top: 1px;
+`;
+
+type WithConditionalActionProps = {
+    notification: NotificationEntry;
+    header: React.ReactNode;
+    body: React.ReactNode;
+    icon?: JSX.Element;
+    actionLabel: ExtendedMessageDescriptor['id'];
+    actionCondition: {
+        path?: string;
+        network?: Network['symbol'];
+    };
+    onAction: Action;
+    onCancel: Action;
+};
+
+const withConditionalAction = (
+    View: React.ComponentType<ViewProps>,
+    props: WithConditionalActionProps,
+) => {
+    const WrappedView = connect()(({ dispatch }: { dispatch: Dispatch }) => {
```

Comment:
Not sure, but `connect()` is probably some old ugly syntax we do not use anymore?

Reply:
right! it should be replaced by `useActions` hook

Reply:
Will be solved in following PR https://github.com/trezor/trezor-suite/pull/4589.